File: app/src/main/java/com/fed/notes/database/Note.java
package com.fed.notes.database;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.util.Date;
import java.util.UUID;

/**
 * Created by f on 05.05.2017.
 */

@Entity(tableName = "notes")
public class Note {
    @PrimaryKey
    @NonNull
    public UUID id;
    public String title;
    public String description;
    public Date date;

    public Note() {
        this(UUID.randomUUID());
    }

    private Note(@NonNull UUID id) {
        this.id = id;
        date = new Date();
    }

    public String getPhotoFilename() {
        return "IMG_" + id.toString() + ".jpg";
    }
}

File: app/src/main/java/com/fed/notes/database/Note.kt
package com.fed.notes.database

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import java.util.Date
import java.util.UUID

/**
 * Created by f on 05.05.2017.
 */

@Entity(tableName = "notes")
class Note private constructor(@field:PrimaryKey
                               var id: UUID) {
    constructor() : this(UUID.randomUUID())

    var title: String? = null
    var description: String? = null
    var date: Date = Date()
    val photoFilename: String
        get() = "IMG_" + id.toString() + ".jpg"
}